### PR TITLE
Add readonly api to gateway

### PR DESCRIPTION
**Pull request by rht**
Based on https://github.com/ipfs/go-ipfs/pull/1389

(wip)

### CODE REVIEW

**Comment by jbenet** [comment]
cc @harlantwood as he'll be interested to follow this

**Comment by rht** [comment]
The remaining todo here is to write test cases.

**Comment by harlantwood** [comment]
Great to see this moving!  We're doing interesting work on the graph visualization side... Will share more soon.

**Comment by rht** [comment]
@jbenet are the test cases adequate? If so, RFM. Otherwise testing all the API commands might be redundant.

**Comment by jbenet** [comment]
@rht looks great to me! so simple :) :+1:

**Comment by jbenet** [comment]
@harlantwood @krl read only api is in! :+1: thanks @rht

**Comment by harlantwood** [comment]
Awesome thanks @rht!

**Comment by harlantwood** [comment]
@jbenet I'm interested to know when this is released to the public gateway.  Is there a way we track what is released there?  Otherwise maybe comment here when it's live.

**Comment by jbenet** [comment]
@lgierth could you deploy to public gateways?



—

**Comment by whyrusleeping** [comment]
:+1:

File: core/commands/root.go [hunk 1]
```diff
@@ -109,9 +109,39 @@ var rootSubcommands = map[string]*cmds.Command{
...
 }
+var rootROSubcommands = map[string]*cmds.Command{
+	"block": &cmds.Command{
+		Subcommands: map[string]*cmds.Command{
+			"stat": blockStatCmd,
+			"get":  blockGetCmd,
+		},
+	},
+	"cat":      CatCmd,
+	"commands": CommandsDaemonCmd,
```
Comment (rht) [inline core/commands/root.go]: Wait, one more small fix: this lists the Root commands instead of RootRO.

File: core/commands/root.go [hunk 2]
```diff
@@ -109,9 +109,39 @@ var rootSubcommands = map[string]*cmds.Command{
...
+	},
+	"object": &cmds.Command{
+		Subcommands: map[string]*cmds.Command{
+			"data":  objectDataCmd,
+			"links": objectLinksCmd,
+			"get":   objectGetCmd,
+			"stat":  objectStatCmd,
+		},
+	},
+	"refs": RefsCmd,
```
Comment (rht) [inline core/commands/root.go]: One more here: refs has `refs local` to list all the local hashes.

**Comment by rht** [comment]
API is sanitized here https://github.com/ipfs/go-ipfs/pull/1583

**Pull request merged**